code/game/objects/items/weapons/stunbaton.dm

```diff
@@ -181,22 +179,32 @@
 			"<span class='danger'>[L == user ? "You prod yourself" : "[user] has prodded you"] with [src]. Luckily it was off.</span>")
 		return
 
-	baton_stun(L, user)
-	user.do_attack_animation(L)
+	if(baton_stun(L, user))
+		user.do_attack_animation(L)
+
+/obj/item/melee/baton/proc/baton_stun(mob/living/L, mob/user, skip_cooldown = FALSE)
+	if(cooldown > world.time && !skip_cooldown)
+		return FALSE
+
+	var/user_UID = user.UID()
+	if(HAS_TRAIT_FROM(L, TRAIT_WAS_BATONNED, user_UID)) // prevents double baton cheese.
+		return FALSE
 
-/obj/item/melee/baton/proc/baton_stun(mob/living/L, mob/user)
+	cooldown = world.time + initial(cooldown)
```

Comment:
Personal preference of course, but i feel this kind of cooldown advancement is a "clever hack" and makes the code much harder to read.
`cooldown` is documented to be "`/// the amount of time between swings`". But in reality that's only the semantics of the *initial* cooldown value, and if you try to vv the thing in game, you would be pretty confused (and also unable to adminbus a no-cooldown baton).

I think it's much better to create an extra variable `var/cooldown_expires` or `var/cools_down_at` or some such, that would hold the time it's usable again, instead of reusing the same `cooldown` for both "duration" and "moment in time".